rskj-core/src/main/java/org/ethereum/core/Transaction.java

```diff
@@ -99,13 +105,19 @@ public class Transaction {
     private Keccak256 hash;
     private Keccak256 rawHash;
 
-    protected Transaction(byte[] rawData) {
-        this(RLP.decodeList(rawData));
-    }
+    //EIP-2718
+    private byte type;
+    //Account Abstraction - Can be of any type, not only ECDSA
+    private byte[] rawsignature;
+
 
-    protected Transaction(RLPList transaction) {
+    protected Transaction(byte[] rawData) {
+        final boolean typed = Hex.toHexString(new byte[]{rawData[0]}).compareTo(TYPED) < 0;
+        this.type = typed ? rawData[0] : LEGACY_TYPE;
+        final byte[] data = typed ? Arrays.copyOfRange(rawData,1, rawData.length) : rawData;
+        RLPList transaction = RLP.decodeList(data);
         if (transaction.size() != 9) {
-            throw new IllegalArgumentException("A transaction must have exactly 9 elements");
```

Comment:
alternative is to keep the fields v, r, s as is and to add new fields rawSig and from. This will make the elementsmore than 9